openapi.yaml

```diff
@@ -0,0 +1,2411 @@
+          type: string
+          readOnly: true
+        description:
+          type: string
+          readOnly: true
+        concurrency:
+          type: number
+          readOnly: true
+        start_date:
+          type: string
+          format: 'date-time'
+          readOnly: true
+        dag_run_timeout:
+          $ref: '#/components/schemas/TimeDelta'
+        doc_md:
+          type: string
+          readOnly: true
+        default_view:
+          type: string
+          readOnly: true
+        tags:
+          type: array
+          readOnly: true
+          items:
+            $ref: '#/components/schemas/Tag'
+
+    ExtraLink:
+      type: object
+      properties:
+        class_ref:
+          $ref: '#/components/schemas/ClassReference'
+        name:
+          type: string
+          readOnly: true
+        href:
+          type: string
+          readOnly: true
+
+    ExtraLinkCollection:
+      type: object
+      properties:
+        extra_links:
+          type: array
+          items:
+            $ref: '#/components/schemas/ExtraLink'
+
+    Task:
+      type: object
+      properties:
+        class_ref:
+          $ref: '#/components/schemas/ClassReference'
+        task_id:
+          type: string
+          readOnly: true
+        owner:
+          type: string
+          readOnly: true
+        start_date:
+          type: string
+          format: 'date-time'
+          readOnly: true
+        end_date:
+          type: string
+          format: 'date-time'
+          readOnly: true
+        trigger_rule:
+          $ref: '#/components/schemas/TriggerRule'
+        extra_links:
+          type: array
+          readOnly: true
+          items:
+            type: object
+            properties:
+              class_ref:
+                $ref: '#/components/schemas/ClassReference'
+        depends_on_past:
+          type: boolean
+          readOnly: true
+        wait_for_downstream:
+          type: boolean
+          readOnly: true
+        retries:
+          type: number
+          readOnly: true
+        queue:
+          type: string
+          readOnly: true
+        pool:
+          type: string
+          readOnly: true
+        pool_slots:
+          type: number
+          readOnly: true
+        execution_timeout:
+          $ref: '#/components/schemas/TimeDelta'
+        retry_delay:
+          $ref: '#/components/schemas/TimeDelta'
+        retry_exponential_backoff:
+          type: boolean
+          readOnly: true
+        priority_weight:
+          type: number
+          readOnly: true
+        weight_rule:
+          $ref: '#/components/schemas/WeightRule'
+        ui_color:
+          $ref: '#/components/schemas/Color'
+        ui_fgcolor:
+          $ref: '#/components/schemas/Color'
+        template_fields:
+          type: array
+          readOnly: true
+          items:
+            type: string
+        sub_dag_id:
+          type: string
+          readOnly: true
```

Comment:
The `readOnly` flag? That tells OpenAPI that this comes in the response, but that it is not editable by PATCH methods https://swagger.io/docs/specification/data-models/data-types/#readonly-writeonly

Reply:
No, the `sub_dag_id` bit

Reply:
```
# subdag parameter is only set for SubDagOperator.
# Setting it to None by default as other Operators do not have that field
```
https://github.com/apache/airflow/blob/14fb58530c219e5bd9df373c65a88bea8ca055c9/airflow/models/baseoperator.py#L430-L433
https://github.com/apache/airflow/blob/14fb58530c219e5bd9df373c65a88bea8ca055c9/airflow/serialization/schema.json#L152

Reply:
Yes but both SubDagOperator.subdag and "subdag" field in JSON Schema have SubDag itself and not just dag_id.

Reply:
Ohh. Good point.  We can't retrieve  SubDAG without downloading the full DAG because it is a DAG structure, not a DAG Model.
https://github.com/apache/airflow/pull/8721/commits/dc73c66f286e58d383f3584b85684fd24c6d1254